pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/scheduler/SplitScheduler.java

```diff
@@ -0,0 +1,172 @@
+ */
+@Slf4j
+public class SplitScheduler implements LoadManagerScheduler {
+
+    private final PulsarService pulsar;
+
+    private final ScheduledExecutorService loadManagerExecutor;
+
+    private final LoadManagerContext context;
+
+    private final ServiceConfiguration conf;
+
+    private final ServiceUnitStateChannel serviceUnitStateChannel;
+
+    private final NamespaceBundleSplitStrategy bundleSplitStrategy;
+
+    private final SplitCounter counter;
+
+    private final AtomicReference<List<Metrics>> splitMetrics;
+
+    private volatile ScheduledFuture<?> task;
+
+    private long counterLastUpdatedAt = 0;
+
+    public SplitScheduler(PulsarService pulsar,
+                          ServiceUnitStateChannel serviceUnitStateChannel,
+                          SplitCounter counter,
+                          AtomicReference<List<Metrics>> splitMetrics,
+                          LoadManagerContext context,
+                          NamespaceBundleSplitStrategy bundleSplitStrategy) {
+        this.pulsar = pulsar;
+        this.loadManagerExecutor = pulsar.getLoadManagerExecutor();
+        this.counter = counter;
+        this.splitMetrics = splitMetrics;
+        this.context = context;
+        this.conf = pulsar.getConfiguration();
+        this.bundleSplitStrategy = bundleSplitStrategy;
+        this.serviceUnitStateChannel = serviceUnitStateChannel;
+    }
+
+    public SplitScheduler(PulsarService pulsar,
+                          ServiceUnitStateChannel serviceUnitStateChannel,
+                          SplitCounter counter,
+                          AtomicReference<List<Metrics>> splitMetrics,
+                          LoadManagerContext context) {
+        this(pulsar, serviceUnitStateChannel, counter, splitMetrics, context,
+                new DefaultNamespaceBundleSplitStrategyImpl(counter));
+    }
+
+    @Override
+    public void execute() {
+        boolean debugMode = conf.isLoadBalancerDebugModeEnabled() || log.isDebugEnabled();
+        if (debugMode) {
+            log.info("Load balancer enabled: {}, Split enabled: {}.",
+                    conf.isLoadBalancerEnabled(), conf.isLoadBalancerAutoBundleSplitEnabled());
+        }
+
+        if (!isLoadBalancerAutoBundleSplitEnabled()) {
+            if (debugMode) {
+                log.info("The load balancer or load balancer split already disabled. Skipping.");
+            }
+            return;
+        }
+
+        synchronized (bundleSplitStrategy) {
+            final Set<SplitDecision> decisions = bundleSplitStrategy.findBundlesToSplit(context, pulsar);
+            if (!decisions.isEmpty()) {
+                List<CompletableFuture<Void>> futures = new ArrayList<>();
+                for (SplitDecision decision : decisions) {
+                    if (decision.getLabel() == Success) {
+                        var split = decision.getSplit();
+                        futures.add(serviceUnitStateChannel.publishSplitEventAsync(split)
```

Comment:
Do we need to wait until the split (Received `Deleted` message) is finished,

Reply:
Great point. I see that UnloadScheduler does not wait for unload completion now.

However, I agree that we better wait for completion (wait on channel.getOwnerAsync) to confirm the completion on the same dedicated thread.

The class name, SplitScheduler would sound counterintuitive if the logic schedules splits and waits for completion. Perhaps, the better name could be SplitManager.

Do we want this class name change and adding the waiting logic in this PR, or a separate PR(also refactoring UnloadScheduler)?

Reply:
How about we add a method like this? 
https://github.com/apache/pulsar/pull/19538/files#diff-b9a27acaccf9d79fd9c1e344f940ade6565f3c1b9fa8000e62d0c116e311a113R134

Reply:
Let's continue this discussion in your PR, as you have a proposal for this waiting logic in that PR.